lib/letsencrypt.js

```diff
@@ -1,215 +1,104 @@
+
+                const pkg = require('../package.json');
+                const gl = require('greenlock').create({
+                    packageRoot: __dirname,
+                    configDir: leDir,
+                    maintainerEmail: leSettings.email,
+                    cluster: false,
+                    packageAgent: pkg.name + '/' + pkg.version,
+                    notify: function (event, details) {
+                        // TODO: separate out errors, etc.
+                        log.debug(`Greenlock notify ${event}: ${details}`);
+                    }
+                });
+                gl.manager.defaults({
+                    agreeToTerms: true,
+                    subscriberEmail: leSettings.email
+                }).then(() => {
+                    // Add sites.
+                    // TODO: something seems fishy with the timing of this because on the first run
+                    // a warning about no sites being configured is found, even though things go on to
+                    // work perfectly.
+                    for (const domain of leSettings.domains) {
+                        log.debug(`Adding to Greenlock: ${domain}`);
+                        gl.add({ subject: domain, altnames: [domain] });
+                    }
 
-            try {
-                server = require('https').createServer(options, lex.middleware(app));
-            } catch (err) {
-                log.error('HTTPS server could not be started: ' + err);
-            }
-        } else {
-            try {
-                server = require('https').createServer(certificates, app);
-            } catch (err) {
-                log.error('HTTPS server could not be started: ' + err);
+                    require('greenlock-express')
+                        .init({
+                            greenlock: gl
+                        }).ready(glx => {
+                            // The below is from greenlock-express examples/https/server.js
+
+                            // Start the challenge server if configured
+                            if (settings.leUpdate) {
+                                const httpServer = glx.httpServer();
+                                settings.lePort = parseInt(settings.lePort, 10) || 80;
+                                const bind = (!settings.bind || settings.bind === '0.0.0.0') ? undefined : settings.bind || undefined;
+                                httpServer.listen(settings.lePort, bind, function () {
+                                    console.info('Listening on ', httpServer.address());
+                                });
+                                log.info('Challenge server is started on ' + settings.lePort);
+                            }
+
+                            // Don't call listen here because caller will do that
+                            resolve(glx.httpsServer(null, app));
```

Comment:
There was never a way to configure the HTTPS port, right?

Reply:
There is, but the port listen is done by the caller. This is how it used to work.

Reply:
OK, perfect.